ClassLibrary/ClassLibrary/RepositoryBooks.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace ClassLibrary
+{
+    public class RepositoryBooks
+    {
+        // looks like with "DepartmentReaders", you can write for one occasion...  
+
+        //for comfortable I do it open, but I break AV1014 (An object should not expose any other classes it depends on for Library)
+        private List<Book> Books
+        {
+            get;
+            set;
+        }
+            = new List<Book>();
+
+        public void AddBook(Book book)
+        {
+            if (book == null)
+            {
+                throw new ArgumentNullException("Book's reference is null");
+            }
+
+            Book foundBook = Books?.FirstOrDefault(x => x.ISBN == book.ISBN);
+            if (foundBook != null)
+            {
+                throw new ArgumentException($"This book's [{book.ISBN},{book.NameBook}] exists");
+            }
+            else
+            {
+                Books.Add(book);
+            }
+        }
+
+        // what should we return ?
+        // what AV1130? I return "list" (For convenience), but use IEnumerable inside 
+        // And how should I convert the list (just below)? 
+        public List<Book> FindBooks(String NameBook="",String NameAutor="",String LastNameAutor="")
```

Comment:
There are two problems with returning a reference type:
1. calling code can modify the returned collection, so if that's an internal collection and not a copy - the internal state of the called code will change, that's extremely error-prone (AV1130 is about that);
2. called code can modify the collection after returning it, so calling code might get different results for subsequent reads - that's also extremely error-prone;

To fix those:
1. return type should not allow modification (for example, `IEnumerable<T>` instead of `List<T>`)
2. return value should be a copy (ideally deep copy) of the internal object (for example, `return Books.ToList()` will make a copy)